### ISSUE TITLE
Adding Support of supplying .Forwards file in genapi

### ISSUE THREAD

**Issue by Anipik** [dotnet/arcade#5807]
Some of the refs also contains typeforwards. 
Genapi adds this typeforwards with all the members  to the contractAssemblyName.cs.
The solution here will be automatically look for typeforwards file in the ref folder and parse the them to include these type to api exclude type list.

cc @ericstj @ViktorHofer @ViktorHofer

**Comment by ViktorHofer** [comment]
cc @safern @joperezr

**Comment by safern** [comment]
FWIW we have support in dotnet/runtime for a file named: `ReferenceSourceExcludeApi.txt` which we pass down to GenAPI in cases where you have .Manual.cs files to avoid having to remove them from the generated reference source.

https://github.com/dotnet/runtime/blob/52d4d8c057d24a77b2cf6708d495cf549a8bee9a/src/libraries/System.Reflection.Metadata/ref/ReferenceSourceExcludeApi.txt

In order to avoid doing this, should we also look for .Manual.cs file?

**Comment by Anipik** [comment]
yeah the new code will parsing typeforwards into docids, rest everything is already there.

> In order to avoid doing this, should we also look for .Manual.cs file?

adding support for this could not be tough, we already do this in genfacades i.e. converting types in docIds.

**Comment by safern** [comment]
Right, that's what I mean, this issue should cover both, `TypeForwards` and `Manual`. If we're already doing the job for `TypeForwards` we should probably use the same train for `Manual` 😄

**Comment by ViktorHofer** [comment]
Closing as GenAPI got rewritten and now lives in dotnet/sdk: https://github.com/dotnet/sdk/tree/main/src/Compatibility
